pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -434,30 +443,49 @@ cdef class {{name}}HashTable(HashTable):
             'upper_bound' : self.table.upper_bound,
         }
 
-    cpdef get_item(self, {{dtype}}_t val):
+    cpdef get_item(self, {{dtype}}_t val, bint na_value = False):
```

Comment:
Would something like `val_is_na` be a more descriptive keyword name? (typically in other places where we have `na_value`, it is the value itself, not a bool flag)

(while you are at it, a brief docstring could also help to describe the keywords)

Reply:
imo val_is_na is a bit confusing, because ``val`` is actually a placeholder and therefore garbage at this place. It is more like get_item of na_value

Reply:
Although with `get_na_value`, it is giving some duplication (`get_item(.., get_na_value=True)`

But this is also pointing to that it's not a very clean API, with basically two separate behaviours (`value` being garbarge if the other keyword is set). It's not too important given this is deep into the internals, but maybe a clearer interface is to just have two separate methods instead of the keyword in the single method: `get_item(value)` and `get_na()`. 
Since the caller needs to handle this specifically in the level up anyway to check for NA and set the keyword, you might as well just call a different method instead of passing a keyword.

Reply:
Refactored into own methods, should be significantly easier to understand now